src/app/components/home/home.component.ts

```diff
@@ -43,25 +67,20 @@ export class HomeComponent implements OnInit, AfterViewInit, OnDestroy {
       rxMap(ids => loOrderBy(ids, undefined, 'desc'))
     ).subscribe(ids => {
       this.iterationIds = ids;
+      this.updateSubject.next();
       this.changeDetector.detectChanges();
     });
   }
 
   ngAfterViewInit() {
-    setTimeout(() => {
-      const { carousel, route: { firstChild: childRoute } } = this;
-      if (childRoute) {
-        const id = +childRoute.snapshot.paramMap.get('iid');
-        const index = loIndexOf(this.iterationIds, id);
-        if (index >= 0) { carousel.slideTo(index, 0); }
-      }
-    }, 0);
+    this.initSubject.next();
+    this.initSubject.complete();
   }
 
   ngOnDestroy() {
-    if (this.dataSubscription) {
-      this.dataSubscription.unsubscribe();
-    }
+    this.updateSubject.complete();
+    this.dataSubscription.unsubscribe();
+    this.routerSubscription.unsubscribe();
```

Comment:
check if it is defined, then unsubscribe ?

Reply:
No need it is always created in ngOnInit

Reply:
Helps in unit testing plus its a good practice. Always better to have checks.